src/DependencyInjection/Compiler/BrokerPass.php

```diff
@@ -31,15 +31,17 @@ public function process(ContainerBuilder $container)
             }
         }
 
-        // a bit copied from the extension, but no choice here to retrieve the config
-        // without going through parameters...
-        $configs = $container->getExtensionConfig('wisembly_amqp');
-        $config = $container->getExtension('wisembly_amqp')->getConfig($configs);
+        if (false === $container->hasParameter('wisembly.amqp.config.broker')) {
+            throw new InvalidArgumentException('The expected container parameter "wisembly.amqp.config.broker" is missing');
+        }
+
+        $broker = $container->getParameter('wisembly.amqp.config.broker');
+        $container->getParameterBag()->remove('wisembly.amqp.config.broker');
```

Comment:
remove the removing, imo a compiler pass should not remove parameters

Reply:
too bad that a parameter can't be private though... was sure there was a rfc for that but whatever